### PR TITLE
DRYing up checker code

### PR DESCRIPTION
**Pull request by perryqh**
The first 20ish lines of our `check` functions contain similar code.

This PR introduces a PackChecker for code reuse.

### CODE REVIEW

File: src/packs/parsing/ruby/namespace_calculator.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ pub(crate) fn calculate_module_nesting(
             format!("{}::{}", previous, namespace)
         };
 
-        previous = new_nesting.to_owned();
+        new_nesting.clone_into(&mut previous);
```
Comment (alexevanczuk) [inline src/packs/parsing/ruby/namespace_calculator.rs:32]: Why this change?

File: src/packs/pack_set.rs [hunk 2]
```diff
@@ -64,7 +64,7 @@ impl PackSet {
 
         let indexed_packs = indexed_packs_by_name;
 
-        if indexed_packs.get(".").is_none() {
+        if !indexed_packs.contains_key(".") {
```
Comment (alexevanczuk) [inline src/packs/pack_set.rs:67]: Why this change?

File: src/packs/walk_directory.rs [hunk 3]
```diff
@@ -108,8 +108,10 @@ pub(crate) fn walk_directory(
                 children.iter_mut().for_each(|child_dir_entry_result| {
                     if let Ok(child_dir_entry) = child_dir_entry_result {
                         let child_absolute_dirname = child_dir_entry.path();
-                        child_dir_entry.client_state.current_package_yml =
-                            read_dir_state.current_package_yml.clone();
+                        child_dir_entry
```
Comment (alexevanczuk) [inline src/packs/walk_directory.rs:111]: Why this change?

**Review by alexevanczuk — APPROVED** [review]
Nice tidy!

File: src/packs/parsing/ruby/namespace_calculator.rs (hunk 1 above)
Comment (perryqh) [reply src/packs/parsing/ruby/namespace_calculator.rs:32]: My bad, I should have commented on this. A new version of clippy suggested it.

File: src/packs/pack_set.rs (hunk 2 above)
Comment (perryqh) [reply src/packs/pack_set.rs:67]: New clippy
